### PR TITLE
Implemented smart routing

### PR DESCRIPTION
**Pull request by GhostInAMachine**

### CODE REVIEW

File: src/HazelcastClient.ts [hunk 1]
```diff
@@ -7,39 +7,42 @@ import Q = require('q');
...
     public static newHazelcastClient(config?: ClientConfig): Q.Promise<HazelcastClient> {
         var client: HazelcastClient = new HazelcastClient(config);
-        var clientPromise: Q.Promise<HazelcastClient> = client.init();
-        return clientPromise;
+        return client.init();
     }
 
     constructor(config?: ClientConfig) {
-        if (!config) {
-            this.config = new ClientConfig();
```
Comment (mustafaiman) [inline src/HazelcastClient.ts]: What if client does not provide a config? How do we provide default configuration?

Comment (mustafaiman) [inline src/HazelcastClient.ts]: Avoid my comment. I see field initialization now.

**Comment by mustafaiman** [comment]
:+1:

**Comment by asimarslan** [comment]
:+1:

**Pull request merged**